tests/resource_name_test.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/resource_name.h>
+#include <aws/testing/aws_test_harness.h>
+
```

Comment:
how about a negative test case on nested ":"?

as in:

arn::bla:h::helad etc...

Reply:
Is this not a valid ARN where:

partition = ""
service = "bla"
region = "h"
account_id = ""
resource_id = "helad"

While this is quite likely forever not a real ARN, it doesn't seem like the CRT should have logic that understand valid values in each part of an ARN other than the prefix.

Reply:
I believe it is invalid for account_id to be empty, right? Could we just validate emptiness?